### ISSUE TITLE
Permissions problems for all decisions

### ISSUE THREAD

**Issue by rastapopougros** [loomio/loomio#4136]
I can't understand the logic for write permissions. I discern "read permission" and "write permission".

- When a discussion (group or subgroup) is public, all people can **read**, but only people IN this group or subgroup can **write** comments => I'm almost ok with that.
- But in the decisions block : all people *even anonymous* can **create a new decision** : it's a write permission => WTF ?
- But when a decision is already there (by a member of the group) : all people *even anonymous* can **vote in this sondage/referendum/etc** : it's a write permission => WTF ?

When I re-read options, it's in total contradiction with what I checked :
:heavy_check_mark: Members can create propositions => checked, so **only members** should be able to do that, not anyone
:heavy_check_mark: Members can vote => checked, so **only members** should be able to vote, not anyone

If we create groups, and we want to discuss inside, we sometimes want that other people outside can read, *for transparency,* but only members of the group should be able to anwser consultations if we checked "members can…".

(I also know that sometimes, we want to open a consultation to really everybody, it must be an option. But it seems there is a lack of configuration for decisions permissions. It's an other problem, less important.)

In short, at least for current options, it's a bug, because the two behaviors is not what we checked.

**Comment by gdpelican** [comment]
For clarity, the intended behaviour is:

- Polls created inside a group are default 'anyone can participate: false'. This means only group members and invited guests may vote. The poll may be seen by anyone who can see the group (so, for public groups, the poll is still visible)
- Polls created outside a group are default 'anyone can participate: true'. This means you can create a poll and share the link to invite people to participate.
- If you set a group poll to 'anyone can participate: true', then anyone can vote who has the link
- If you set a non-group poll to anyone can participate: false', then only invited guests may vote on the poll.

I think the bug you're describing is that the decision tools card is appearing for people who can't start polls (although it wouldn't actually let them start a poll), which I've just fixed and will be deployed shortly.

**Comment by rastapopougros** [comment]
It's not just the card with the list which is visible : when outside people (non members + anonymous) click on each tool, the box open and they can actually fill the form.

For your points, I don't see what do you mean with "anyone can participate: true|false" because I don't see any option like this poll by poll, nor in the discussion options, nor in the group options.
I have: 
- no permission options in the creation/modification form for a poll
- an option in the discussion "public or private", and it says, if public, that **only members can participate** (yeah it's what it should be, but it's not, not just for the creation list, all people can vote on all poll)
- some options in the group (in "advanced") and it says "**members** can create" or "**members** can participate", no options with "anyone"

(I think it's 1.4.0, https://framavox.org and https://framagit.org/framasoft/framavox )

**Comment by gdpelican** [comment]
This should be fixed in latest. Please ensure that framavox.org (which we have no affiliation with) is running that latest version of Loomio, and if you're still finding an issue please file a report with expected behaviour, observed behaviour, and reproduction steps so that I can more accurately gauge what's wrong.

**Comment by gdpelican** [comment]
Ah, the anyone can participate option appears on the 'share' modal which appears immediately after creating a poll... if it is set to true, anyone will be able to vote on it.

**Comment by rastapopougros** [comment]
Thanks for all those infos!

I opened the "share" box in the menu, and I unchecked "anyone can participate". I saw a confirmation notification and when I re-open, it's always unchecked, ok.

But now when I open the same discussion in a private window (so anonymous): I can see the poll **and vote**. Here is an example: 
https://framavox.org/d/2cR4EDWl/charte-mode-d-emploi-livret-d-accueil

**Comment by gdpelican** [comment]
Framavox appears to be running 1.4.3; I just released 1.7.0, in which this should be fixed.

Closing this now, please reopen if you can repro on loomio.org or another instance running the latest.

**Comment by rastapopougros** [comment]
Thanks! I ping the Framasoft association for the update.

**Comment by rastapopougros** [comment]
Hello, since this morning, the Framavox instance is not anymore in 1.4 but in version 1.9.

And… the problems, or at least some of the problems are still here.

Example for the **creation** of propositions : 
- In the configuration for the group the checkbox "**Members** can create propositions" is checked.
- Expected : **only members** should be able to do that.
- Observed : everyone, *even totally anonymous* can create propositions in our groups.

Look at the right column anonymously :
https://framavox.org/g/tgpSdAEU/spip
